core/src/main/java/zipkin2/reporter/internal/CountBoundedQueue.java

```diff
@@ -102,7 +111,13 @@ int doDrain(SpanWithSizeConsumer<S> consumer) {
       S next = elements[readPos];
 
       if (next == null) break;
-      if (consumer.offer(next, 0)) {
+
+      int nextSizeInBytes = encoder.sizeInBytes(next);
+      int messageSizeOfNextSpan = sender.messageSizeInBytes(nextSizeInBytes);
+      metrics.incrementSpanBytes(nextSizeInBytes);
+
+      if (messageSizeOfNextSpan > messageMaxBytes) break;
```

Comment:
I think we should drop and continue, not break (because this is a span that should have been dropped early, and the next will possibly fit in the same message).

Also, we should increment spans dropped before continue `metrics.incrementSpansDropped(1);`

Reply:
That would perfectly makes sense!